src/addons/messagelog/src/main/java/ee/ria/xroad/proxy/messagelog/TimestamperUtil.java

```diff
@@ -87,6 +87,8 @@ static InputStream makeTsRequest(TimeStampRequest req, String tspUrl) throws Exc
         if (con.getResponseCode() != HttpURLConnection.HTTP_OK) {
             throw new RuntimeException("Received HTTP error: " + con.getResponseCode() + " - "
                     + con.getResponseMessage());
+        } else if (con.getInputStream() == null) {
+            throw new RuntimeException("Could not get response from TSP");
```

Comment:
Throwing generic exceptions is a code smell, e.g. IOException would be more appropriate here.  One should also clean up the connection in case of failure.

Reply:
Fixed in 0f9258c